### PR TITLE
[AIRFLOW-3103][AIRFLOW-3147] Update flask-appbuilder

### PR DESCRIPTION
**Pull request by jmcarp**
Note: depends on https://github.com/dpgaspar/Flask-AppBuilder/pull/811 being merged and released, at which point we'll also bump `flask-appbuilder` in `setup.py`.

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/issues/AIRFLOW-3103) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3103

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by jmcarp** [comment]
Build failures expected until flask-appbuilder is updated.

**Comment by ashb** [comment]
FAB has been released, am re-running the tests to see if they pass.

**Comment by ashb** [comment]
@jmcarp Looks like something else needs an update:

```
   AttributeError: <module 'flask_login' from '/app/.tox/py27-backend_mysql-env_docker/local/lib/python2.7/site-packages/flask_login/__init__.pyc'> does not have the attribute '_get_user'
```

Or I can pick this up if you don't have time. 

Turns out it wasn't that big a change needed. And since you ticked the "allow maintainters to update this branch" I could fix it for you.

File: setup.py [hunk 1]
```diff
@@ -298,10 +298,10 @@ def do_setup():
             'croniter>=0.3.17, <0.4',
             'dill>=0.2.2, <0.3',
             'flask>=0.12.4, <0.13',
-            'flask-appbuilder>=1.11.1, <2.0.0',
+            'flask-appbuilder>=1.12, <2.0.0',
```
Comment (ashb) [inline setup.py:301]: Could you add `[AIRFLOW-3147]` in to your commit message too please?

File: tests/www/test_utils.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def some_func():
                 self.assertEqual(fake_username, kwargs['owner'])
                 mocked_session_instance.add.assert_called_once()
 
-    @mock.patch("flask_login._get_user")
+    @mock.patch("flask_login.utils._get_user")
```
Comment (ashb) [inline tests/www/test_utils.py:146]: Thanks -I see what I did wrong and came to fix it to find you'd done it already :)

**Comment by ashb** [comment]
You need this too:

```
diff --git a/tests/www_rbac/test_security.py b/tests/www_rbac/test_security.py
index 67ea5b30..6e0b5726 100644
--- a/tests/www_rbac/test_security.py
+++ b/tests/www_rbac/test_security.py
@@ -109,7 +109,7 @@ class TestSecurity(unittest.TestCase):

     def test_get_user_roles(self):
         user = mock.MagicMock()
-        user.is_anonymous.return_value = False
+        user.is_anonymous = False
         roles = self.appbuilder.sm.find_role('Admin')
         user.roles = roles
         self.assertEqual(self.security_manager.get_user_roles(user), roles)
@@ -144,7 +144,7 @@ class TestSecurity(unittest.TestCase):
         self.security_manager.init_role(role_name, role_vms, role_perms)
         role = self.security_manager.find_role(role_name)
         user.roles = [role]
-        user.is_anonymous.return_value = False
+        user.is_anonymous = False
         mock_get_all_permissions_views.return_value = {('can_dag_read', 'dag_id')}

         mock_get_user_roles.return_value = [role]
@@ -154,7 +154,7 @@ class TestSecurity(unittest.TestCase):
     @mock.patch('airflow.www_rbac.security.AirflowSecurityManager._has_view_access')
     def test_has_access(self, mock_has_view_access):
         user = mock.MagicMock()
-        user.is_anonymous.return_value = False
+        user.is_anonymous = False
         mock_has_view_access.return_value = True
         self.assertTrue(self.security_manager.has_access('perm', 'view', user))
```

**Comment by feng-tao** [comment]
Cool. Great to see this pr :)

**Review by ashb — CHANGES_REQUESTED** [review]
Could you also please add this section to the UPDATING.md:

```
### Custom auth backends interface change

We have updated the version of flask-login we depend upon, and as a result any
custom auth backends might need a small change: `is_active`,
`is_authenticated`, and `is_anonymous` should now be properties. What this means is if
previously you had this in your user class

    def is_active(self):
      return self.active

then you need to change it like this

    @property
    def is_active(self):
      return self.active

```

**Comment by feng-tao** [comment]
@ashb , I think this pr is ready. And it needs to be checked in to unblock the CI.

**Pull request merged**